router/diag_cntl.h

```diff
@@ -35,6 +35,23 @@
 #include "peripheral.h"
 #include "masks.h"
 
+struct diag_id_tbl_t {
+	struct list_head link;
+	uint8_t diag_id;
+	uint8_t pd_val;
+	uint32_t periph_id;
+	uint8_t pd_feature_mask;
+	char *process_name;
+} __packed;
```

Comment:
I dont think diag_id_tbl_t or diag_id_t needs to be packed. Is there a specific need?